polynote-frontend/polynote/ui/component/notebook/notebookconfig.ts

```diff
@@ -401,15 +401,33 @@ class SparkConf {
 
 }
 
-class EnvConf {
+class KernelConf {
     readonly el: TagElement<"div">;
     private container: TagElement<"div">;
+    private jvmArgsInput: TagElement<"input">;
+    private scalaVersionInput: DropdownElement;
+
+    constructor(configState: StateView<NotebookConfig>) {
+        const envHandler = configState.view("env");
+        const jvmArgsHandler = configState.view("jvmArgs");
+        const scalaVersionHandler = configState.view("scalaVersion");
+
+        // TODO: this could come from the server
+        const availableScalaVersions = {
+            "2.11": "2.11",
+            "2.12": "2.12"
+        }
 
-    constructor(envHandler: StateView<Record<string, string> | undefined>) {
         this.el = div(['notebook-env', 'notebook-config-section'], [
-            h3([], ['Environment Variables']),
-            para([], ['Set environment variables here. Please note this is only supported when kernels are launched as a subprocess (default).']),
-            this.container = div(['env-list'], [])
+            h3([], ['Kernel configuration']),
+            para([], ['Please note this is only supported when kernels are launched as a subprocess (default).']),
+            h4([], 'Scala version:'),
+            para([], `If using Spark, the Scala version must match that of your Spark installation. "Default" will use Polynote's configured Scala version, or auto-detect the appropriate version.`),
+            this.scalaVersionInput = dropdown(['scala-version'], {"": "Default", ...availableScalaVersions}, scalaVersionHandler.state || ""),
+            h4([], 'Environment variables:'),
+            this.container = div(['env-list'], []),
+            para([], ['Additional JVM arguments:']),
+            this.jvmArgsInput = textbox(['jvm-args'], "JVM Arguments", jvmArgsHandler.state).attr("size", "64")
```

Comment:
Might be good to specify that they should be in the command line format, e.g., `-Dfoo=bar`

Reply:
Good point – also it should be an `h4`. Fixed.